packages/shared/lib/services/connections/credentials/refresh.ts

```diff
@@ -62,6 +62,12 @@ interface RefreshProps {
 
 const logger = getLogger('connectionRefresh');
 
+// In-memory cache for in-flight refresh operations to collapse concurrent requests
+const inFlightRefreshes = new Map<
+    string,
+    Promise<Result<{ connection: DBConnectionDecrypted; refreshed: boolean; credentials: RefreshableCredentials }, NangoInternalError>>
+>();
```

Comment:
I am planning to implement a simple fixed size map util to use here and in account.service.ts to cache the environment. I'll raise a PR then